### ISSUE TITLE
Allow navigation items to stay highlighted when clicked

### ISSUE THREAD

**Issue by houtanb** [mmistakes/minimal-mistakes#1876]
The navigation menu is very nice. One thing I'd like to be able to modify is whether or not a menu item stays highlighted when clicked.

Say for example that my menu has two items, About and News:
![image](https://user-images.githubusercontent.com/584723/46544876-0b63c300-c8c5-11e8-9ddf-2ba921ab2f33.png)
If my mouse hovers over News for example, I get:
![image](https://user-images.githubusercontent.com/584723/46544857-fb4be380-c8c4-11e8-8a3e-a5bf4b0e7f23.png)
It'd be nice to have News stay highlighted (potentially with a different color) if it were clicked. This way we wouldn't be required to have a News title on the News page; this information would be evident from the highlighted Navigation menu item.

**Comment by mmistakes** [comment]
This is a layer of complexity I'm not comfortable adding into the core theme. The build times would take a performance hit to do it since it would have to evaluate the current page to apply an "active" state to the menu item.

If you search around there are some blog posts/tutorials that discuss how to do it. Essentially you add some Liquid logic to check the current `page.url`, and if it matches the URL in the menu item apply a `class="active"`... which would then need the appropriate CSS to style it with the underline.

**Comment by hatpick** [comment]
You are already doing it for the sidebar navigation, why not add it to top navigation as well?

**Comment by andybyte** [comment]
> 
> 
> This is a layer of complexity I'm not comfortable adding into the core theme. The build times would take a performance hit to do it since it would have to evaluate the current page to apply an "active" state to the menu item.
> 
> If you search around there are some blog posts/tutorials that discuss how to do it. Essentially you add some Liquid logic to check the current `page.url`, and if it matches the URL in the menu item apply a `class="active"`... which would then need the appropriate CSS to style it with the underline.

I can't find anyway to compare with page.url within the masthead html and I don't know what blog posts/tutorials show how to do it. I tried many things and got nowhere.

**Comment by mmistakes** [comment]
Jekyll’s official documentation has a really good tutorial on how to build a dynamic and nested navigation. It covers how to add active states too. 

https://jekyllrb.com/tutorials/navigation/#scenario-6-applying-the-active-class-for-the-current-page

**Comment by andybyte** [comment]
Thanks. Yeah, I eventually figured it out. For anyone arriving here all I did was capture page.url, pass it through the include statement, and then compare the include to the link.id (I used ids I set in the YAML but could do something else). For some reason `link.id contains include.content` did NOT work, but `include.content contains link.id` DID work. I don't know why but persistence paid off.

my snippet `{% if include.content contains link.id %} style="border-bottom:3px solid white;" {% endif %}`

the documentation that clicked for me: https://jekyllrb.com/docs/includes/#passing-parameter-variables-to-includes
